substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrThreadLocal.java

```diff
@@ -95,6 +96,7 @@ public void beforeThreadRun(IsolateThread isolateThread, Thread javaThread) {
         parentThreadId.set(isolateThread, JavaThreads.getParentThreadId(javaThread));
 
         SubstrateJVM.getThreadRepo().registerThread(javaThread);
+        ExecutionSampleEvent.tryToRegisterExecutionSampleEventCallback();
```

Comment:
I think that this should be after emitting the thread start event (otherwise, execution sample events could trigger before the thread start event which would be weird).

Reply:
Make sense.